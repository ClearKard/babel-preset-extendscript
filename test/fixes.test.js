const preset = require('../src');
const babel = require('babel-core');

it('wrap conditionals in parentheses', () => {
  const example = `
  function toInteger(value) {
    var result = parseFloat(value), remainder = result % 1;
    return result === result ? remainder ? result - remainder : result : 0;
  }

  toInteger(2.1);
  `;

  const { code } = babel.transform(example, { presets: [preset] });
  expect(code).toMatchSnapshot();
});
