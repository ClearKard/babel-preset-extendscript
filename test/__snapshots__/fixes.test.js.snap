// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wrap conditionals in parentheses 1`] = `
"
function toInteger(value) {
  var result = parseFloat(value),
      remainder = result % 1;
  return result === result ? (remainder ? result - remainder : result) : 0;
}

toInteger(2.1);"
`;
